js/src/otel.ts

```diff
@@ -142,7 +143,7 @@ export class AISpanProcessor {
     }
 
     // Always keep root spans (no parent)
-    if (!span.parentSpanContext) {
+    if (!span.parentSpanContext && !span.parentSpanId) {
```

Comment:
might be useful to leave a small comment about how v2 has `parentSpanContext` and v1 has `parentSpanId`